src/beachbot/robot/robotinterface.py

```diff
@@ -11,8 +20,122 @@ def __init__(self):
         self.platform=None
         self.arm=None
 
+        self.controller=None
+        self.controller_lock = threading.Lock()
```

Comment:
Does this have any overlap with the `RobotController._properties_lock = threading.Lock()` ? If so won't it cause issues that you have two separate locks?